apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -41,7 +43,57 @@ public sealed class SpannerConnectionStringBuilder : DbConnectionStringBuilder
         /// Credentials can be retrieved from a file or obtained interactively.
         /// See Google Cloud documentation for more information.
         /// </summary>
-        public ITokenAccess Credential { get; private set; }
+        public ITokenAccess GetCredential()
+        {
+            // Check the ctor override.
+            if (CredentialOverride != null)
+            {
+                return CredentialOverride;
+            }
+
+            //Calculate it from the CredentialFile argument in the connection string.
+            GoogleCredential credential = null;
+            string jsonFile = CredentialFile;
+            if (!string.IsNullOrEmpty(jsonFile))
+            {
+                if (string.Compare(Path.GetExtension(jsonFile), ".json", StringComparison.OrdinalIgnoreCase) != 0)
+                {
+                    throw new ArgumentException($"{nameof(CredentialFile)} should only be set to a JSON file.");
+                }
+
+                if (!File.Exists(jsonFile))
+                {
+                    //try some relative locations.
+                    jsonFile = $"{GetApplicationFolder()}{Path.DirectorySeparatorChar}{jsonFile}";
+                }
+                if (!File.Exists(jsonFile))
+                {
+                    //throw a meaningful error that tells the developer where we looked.
+                    throw new FileNotFoundException($"Could not find {nameof(CredentialFile)}. Also looked in {jsonFile}.");
+                }
+
+                //TODO(benwu) replace with GoogleCredential.FromFile
+                using (var credStream = File.OpenRead(jsonFile))
+                {
+                    credential = GoogleCredential.FromStream(credStream);
+                    if (credential.IsCreateScopedRequired)
```

Comment:
It's safe to unconditionally call CreateScoped - it'll just return the original credential if it doesn't need scoping. We almost certainly want to move to that pattern elsewhere.

cc @chrisdunelm to fact-check me on that.